### ISSUE TITLE
MIME handling of multipart form data isn't working in several cases

### ISSUE THREAD

**Issue by pshaughn** [servo/servo#25106]
WPT fetch/api/response/response-consume.html fails several cases by getting "Inappropriate MIME-type for Body" thrown by script/body.rs. The comment at https://github.com/servo/servo/blob/master/components/script/body.rs#L187 is probably relevant.

Other cases are failing on ReadableStream, but this seems entirely disjoint from that.

**Comment by CYBAI** [comment]
In case there's any code changes to `body.rs`, just helped to paste its permalink below 👇

https://github.com/servo/servo/blob/64bcce23898891cfdfe6a8b3baa3728aa2f12f27/components/script/body.rs#L187-L197

**Comment by pshaughn** [comment]
Right, any one of these issues could stay in the system a long time so I should remember to do that. Thanks.

**Comment by pshaughn** [comment]
We're using the url crate to parse urlencoded form data; maybe there's just another crate we can use to parse multipart form data? https://docs.rs/multipart/0.2.2/multipart/ might work.

**Comment by jdm** [comment]
Specification: https://fetch.spec.whatwg.org/#ref-for-concept-body-consume-body%E2%91%A2

I think we should add https://crates.io/crates/mime-multipart-hyper1 as a dependency to the script crate and call https://docs.rs/mime-multipart-hyper1/latest/mime_multipart_hyper1/fn.read_multipart_body.html

**Comment by servo-highfive** [comment]
Hi! If you have any questions regarding this issue, feel free to make a comment here, or ask it in [Zulip](https://servo.zulipchat.com/).

If you intend to work on this issue, then add `@servo-highfive: assign me` to your comment, and I'll assign this to you. :smile:
